ProcessOptimizer/plots.py

```diff
@@ -1,2382 +1,2493 @@
+            rvs_ = np.array(sample_points)  # copy
+            # We replace the values in the dimension that we want to keep fixed
+            rvs_[:, dim_locs[i]: dim_locs[i + 1]] = x_
+            # In case of `x_eval=None` rvs conists of random samples.
+            # Calculating the mean of these samples is how partial dependence
+            # is implemented.
+            funcvalue, stddev = model.predict(rvs_, return_std=True)
+            yi.append(np.mean(funcvalue))
+            stddevs.append(np.mean(stddev))
+        # Convert yi and stddevs from lists to numpy arrays
+        yi = np.array(yi)
+        stddevs = np.array(stddevs)
+
+        return xi, yi, stddevs
+
+    else:
+        xi, xi_transformed = space.dimensions[j].evenly_sample(n_points)
+        yi, yi_transformed = space.dimensions[i].evenly_sample(n_points)
+
+        # stddev structure is made regardless of whether it is returned,
+        # since this is cheap and makes the code simpler.
+        zi = []
+        stddev_matrix = []
+        for x_ in xi_transformed:
+            value_row = []
+            stddev_row = []
+            for y_ in yi_transformed:
+                rvs_ = np.array(sample_points)  # copy
+                rvs_[:, dim_locs[j]: dim_locs[j + 1]] = x_
+                rvs_[:, dim_locs[i]: dim_locs[i + 1]] = y_
+                funcvalue, stddev = model.predict(rvs_, return_std=True)
+                value_row.append(np.mean(funcvalue))
+                stddev_row.append(np.mean(stddev))
+            zi.append(value_row)
+            stddev_matrix.append(stddev_row)
+        if return_std:
+            return xi, yi, np.array(zi).T, np.array(stddev_matrix).T
+        else:
+            return xi, yi, np.array(zi).T
+
+
+def plot_objective(
+    result,
+    levels=10,
+    oversampling=None,
+    graph_width=20,
+    size=2,
+    zscale="linear",
+    dimensions=None,
+    usepartialdependence=False,
+    pars="result",
+    expected_minimum_samples=None,
+    title=None,
+    show_confidence=False,
+    plot_options=None,
+):
+    """Pairwise dependence plot of the objective function.
+
+    The diagonal shows the dependence for dimension `i` with
+    respect to the objective function. The off-diagonal shows the
+    dependence for dimensions `i` and `j` with
+    respect to the objective function. The objective function is
+    approximated by `result.model.`
+
+    Pairwise scatter plots of the points at which the objective
+    function was directly evaluated are shown on the off-diagonal.
+    A red point indicates per default the best observed minimum, but
+    this can be changed by changing argument ´pars´.
+
+    Parameters
+    ----------
+    * `result` [`OptimizeResult`]
+        The result for which to create the scatter plot matrix.
+
+    * `levels` [int, default=10]
+        Number of levels to draw on the contour plot, passed directly to
+        `plt.contour()`.
+
+    * `oversampling` [int, default=10]
```

Comment:
The default set in the function definition is None, not 10. Either change it here or in the function default.